src/var/isAttached.js

```diff
@@ -1,11 +1,19 @@
 define( [
 	"../core",
+	"./getShadowRoot",
+	"./document",
 	"../selector" // Get jQuery.contains
-], function( jQuery ) {
+], function( jQuery, getShadowRoot, document ) {
 	"use strict";
 
+	// Replace the use of contains function to check attachment (gh-3504)
 	return function isAttached( obj ) {
+
+		// Check if browser supports Shadow DOM
+		if ( document.head.createShadowRoot || document.head.attachShadow ) {
```

Comment:
`isAttached` is a potentially hot path and I'd rather not see a support check inside of it. How about refactoring getShadowRoot.js to define the smart function conditionally (falling back on `function( elem ) {}`), and then always using it here:
```js
return jQuery.contains( elem.ownerDocument, elem ) ||
	getShadowRoot( elem ) === elem.ownerDocument;
```